modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryServiceImpl.java

```diff
@@ -579,19 +587,77 @@ private void _validateSubmissionLimit(long objectDefinitionId, User user)
 			return;
 		}
 
-		int count = objectEntryPersistence.countByU_ODI(
-			user.getUserId(), objectDefinitionId);
-		long maximumNumberOfGuestUserObjectEntriesPerObjectDefinition =
-			_objectConfiguration.
-				maximumNumberOfGuestUserObjectEntriesPerObjectDefinition();
+		if (FeatureFlagManagerUtil.isEnabled("LPS-192957")) {
+			ObjectDefinition objectDefinition =
+				_objectDefinitionPersistence.findByPrimaryKey(
+					objectDefinitionId);
+
+			try {
+				_objectConfiguration =
+					_configurationProvider.getCompanyConfiguration(
+						ObjectConfiguration.class,
+						objectDefinition.getCompanyId());
+
+				if (_objectConfiguration == null) {
+					_objectConfiguration =
+						_configurationProvider.getSystemConfiguration(
+							ObjectConfiguration.class);
+				}
+			}
+			catch (ConfigurationException configurationException) {
+				throw new RuntimeException(configurationException);
+			}
+
+			LocalDate beginningDate = _getStartDateInterval();
+
+			long count =
+				objectEntryLocalService.getObjectEntriesCount(user.getUserId(),
+					objectDefinition, ObjectEntryTable.INSTANCE.createDate.gte(
+						Date.from(beginningDate.atStartOfDay(
+							ZoneId.systemDefault()).toInstant())));
+
+			long maximumNumberOfGuestUserObjectEntriesPerObjectDefinition =
+				_objectConfiguration.
+					maximumNumberOfGuestUserObjectEntriesPerObjectDefinition();
+
+			if (count >=
+				maximumNumberOfGuestUserObjectEntriesPerObjectDefinition) {
+				throw new ObjectEntryCountException(
+					StringBundler.concat(
+						"Unable to exceed ",
+						maximumNumberOfGuestUserObjectEntriesPerObjectDefinition,
+						" guest object entries for object definition ",
+						objectDefinitionId));
+			}
 
-		if (count >= maximumNumberOfGuestUserObjectEntriesPerObjectDefinition) {
-			throw new ObjectEntryCountException(
-				StringBundler.concat(
-					"Unable to exceed ",
-					maximumNumberOfGuestUserObjectEntriesPerObjectDefinition,
-					" guest object entries for object definition ",
-					objectDefinitionId));
+		}
+		else {
+			int count = objectEntryPersistence.countByU_ODI(
+				user.getUserId(), objectDefinitionId);
+			long maximumNumberOfGuestUserObjectEntriesPerObjectDefinition =
+				_objectConfiguration.
+					maximumNumberOfGuestUserObjectEntriesPerObjectDefinition();
+
+			if (count >=
+				maximumNumberOfGuestUserObjectEntriesPerObjectDefinition) {
+				throw new ObjectEntryCountException(
+					StringBundler.concat(
+						"Unable to exceed ",
+						maximumNumberOfGuestUserObjectEntriesPerObjectDefinition,
+						" guest object entries for object definition ",
+						objectDefinitionId));
+			}
+		}
+	}
+
+	private LocalDate _getStartDateInterval() {
+
+		if (_objectConfiguration.timeScale().equals("days")) {
```

Comment:
Hey @thallescezar , what do you think to change this code from one of these two approaches:

First:
`private LocalDate _getStartDateInterval() {

	if (_objectConfiguration.timeScale().equals("days")) {
		return LocalDate.now().minusDays(_objectConfiguration.duration());
	}

        return LocalDate.now().minusDays(
	        _objectConfiguration.duration() * 7);
}`

Second:
`private LocalDate _getStartDateInterval() {

	if (_objectConfiguration.timeScale().equals("days")) {
		return LocalDate.now().minusDays(_objectConfiguration.duration());
	}
	else if (_objectConfiguration.timeScale().equals("weeks")) {
		return LocalDate.now().minusDays(
			_objectConfiguration.duration() * 7);
	}
}`

I think it's also a good idea to create constants in this class for "days" and "weeks".